src/edit/mouse_events.js

```diff
@@ -305,8 +305,13 @@ function leftButtonSelect(cm, event, start, behavior) {
   function done(e) {
     cm.state.selectingText = false
     counter = Infinity
-    e_preventDefault(e)
-    display.input.focus()
+    // If e is null or undefined we interpret this as someone trying
+    // to explicitly cancel the selection rather than the user
+    // letting go of the mouse button.
+    if (e) {
+      e_preventDefault(e)
+      display.input.focus()
```

Comment:
@marijnh We didn't discuss this in the issue but I felt it'd be weird if `swapDoc` started fiddling with focus all of a sudden. If there's a better way to work around this I'm all ears.

Reply:
Right, that's slightly kludgey but I agree it makes sense, and with the comment it isn't too confusing